python/ray/tests/test_dataclient_disconnect.py

```diff
@@ -1,57 +1,29 @@
-from ray.util.client.ray_client_helpers import ray_start_client_server
-from unittest.mock import Mock
+from ray.util.client.ray_client_helpers import ray_start_client_server_pair
+import time
 import pytest
 
 
-def test_dataclient_disconnect_on_request():
```

Comment:
This test no longe makes sense, since the dataclient no longer fails permanently on a single Rpc error

Reply:
Wait, removing this would mean we have a bug in our reconnection fix? reconnection should not happen always, in the case in this test it should actually fail because it is an application level failure.

Reply:
Restored the original tests with some adjustments to remain compatible with the PR. The main issue before was that the error wasn't really possible to hit in regular operation (i.e. having a Mock in the data request stream), but worked around by adding an explicit check in the retry logic to give up on that case.